flask/helpers.py

```diff
@@ -553,7 +538,9 @@ def send_file(filename_or_fp, mimetype=None, as_attachment=False,
         rv.cache_control.max_age = cache_timeout
         rv.expires = int(time() + cache_timeout)
 
-    if add_etags and filename is not None:
+    if add_etags and filename is not None and file is None:
```

Comment:
Thanks for writing this, there were some glaring errors in this PR. I amended at https://github.com/pallets/flask/pull/1988